### PR TITLE
infra: Migrate Repositories

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
Towards #200:

 * Add `/CODOWNERS`, making @Erethon, @fricklerhandwerk, @Janik-Haag and me owners of `/infra`.
 * Merge the following two repositories into `/infra`, while preserving their history. Both repositories should be archived as soon as this PR is merged.
   * [ngi-nix/infra](https://github.com/ngi-nix/infra)
   * [ngi-nix/ngi0-infra](https://github.com/ngi-nix/ngi0-infra)
 * Remove most of the configuration of the "makemake" machine, in particular everything related to Hydra.
 * Start over with a new configuration.

I have deployed the new machine to a VM, it boots and is happy.

Since the PR diff doesn't really show what changes I made relative to the two previous infra repositories, [here](https://github.com/ngi-nix/ngipkgs/compare/d7d230f..infra)'s a more precise one: `git diff d7d230f..infra -- infra`

@Erethon: Happy to have your review. If you want, we can also set up a call and I walk you through.

@Ericson2314: Once this is deployed, which could be any time without further notice, [hydra.ngi0.nixos.org](https://hydra.ngi0.nixos.org/) will be gone for good. Are you still using it in anyway?

⚠️ Please don't squash merge this. I'd really like to keep the history of the NixOS/infra repo.

### CODE REVIEW

**Comment by lorenzleutgeb** [comment]
On 57ba680 I can still not get through `nix build .#nixosConfigurations.makemake.config.system.build.toplevel` because `boehm-gc-8.2.4 failed with exit code 1 in patchPhase`

**Review by Ericson2314 — APPROVED** [review]
With the latest changes, hydra will no longer be gone. Good!

**Comment by lorenzleutgeb** [comment]
Just activated 73600d970434eab1862a0c030f90d5647e71c42a. @Erethon please verify that you can `ssh root@hydra.ngi0.nixos.org`.

**Comment by Erethon** [comment]
> Just activated [73600d9](https://github.com/ngi-nix/ngipkgs/commit/73600d970434eab1862a0c030f90d5647e71c42a). @Erethon please verify that you can `ssh root@hydra.ngi0.nixos.org`.

I can confirm I have SSH access.

File: infra/ssh-keys.nix
Comment (fricklerhandwerk) [inline infra/ssh-keys.nix:1]: Probably should remove everyone who's neither on the NixOS infra team nor on the NGI project.

Comment (lorenzleutgeb) [reply infra/ssh-keys.nix:1]: Done in 6a71027. To verify:

```sh
$ comm -23 \
  <(curl --silent 'https://raw.githubusercontent.com/NixOS/infra/master/ssh-keys.nix' | nix eval --json --file - | jq -r '.infra[]' | sort) \
  <(nix eval --json --file infra/ssh-keys.nix | jq -r '.[]' | sort)
```

File: flake.lock [hunk 1]
```diff
@@ -38,6 +38,22 @@
         "type": "github"
       }
     },
+    "flake-compat_2": {
```
Comment (wegank) [inline flake.lock:41]: Make this follow `flake-compat`?

**Review by Erethon — APPROVED** [review]
Some of the Nix/flake specific parts go over my head, but the `makemake` configuration makes sense and LGTM. Again, not sure about the Hydra setup but it seems reasonable. I think we can merge this since it's a big PR and continue from here with incremental improvements as (if?) we run any issues/diffs.

File: flake.lock (hunk 1 above)
Comment (lorenzleutgeb) [reply flake.lock:41]: I tried my best but didn't manage to. I'll merge as-is and we'll figure it out in a new PR. Sorry.

**Pull request merged**